src/main/java/com/tutorhub/service/impl/SubmoduleServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package com.tutorhub.service.impl;
+
+import com.tutorhub.model.course.Module;
+import com.tutorhub.model.course.Submodule;
+import com.tutorhub.repository.SubmoduleRepository;
+import com.tutorhub.service.ModuleService;
+import com.tutorhub.service.SubmoduleService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class SubmoduleServiceImpl implements SubmoduleService {
+  private final SubmoduleRepository submoduleRepository;
+  private final ModuleService moduleService;
+
+  @Override
+  public Submodule create(Long courseId, Long moduleId, Submodule submodule) {
+    Module module = moduleService.find(courseId, moduleId);
+    submodule.setModule(module);
```

Comment:
I think you also need to put `submodule` to found `module`

Reply:
Sorry, I don't understand what you mean, could you please clarify

Reply:
Sorry, I saw that you removed a collection of `Submodule` from `Module`

Reply:
Oh, got it, thanks